Carta.py

```diff
@@ -374,13 +439,69 @@ def decideWinner(self):
             self.GPinfo.youWin = True
             self.GPinfo.opponentWin = False
             statement = "Carta.py: decideWinner: No available GrabCard, you win"
+            self.dialogs.append("You Win , Opponent Loses")
+            self.phase = Phase.YOUR_ADJUSTMENT
+            self.numCardsToTransfer += 1
         else:
             self.GPinfo.youWin = False
             self.GPinfo.opponentWin = False
             statement = "Carta.py: decideWinner: Both players grab wrong card"
+            self.dialogs.append("No one Wins")
+            if ((self.GPinfo.yourGrabCardLastWord != self.curReadingCard.getLastWord()) and \
+                  (self.GPinfo.oppoGrabCardLastWord != self.curReadingCard.getLastWord())):
+                self.phase = Phase.OPPONENT_ADJUSTMENT
+                self.numCardsToTransfer += 1
+
+        if (self.phase == Phase.YOUR_ADJUSTMENT):
+            self.dialogs.append("Your Transfer Starts")
+        elif (self.phase == Phase.OPPONENT_ADJUSTMENT):
+            self.dialogs.append("Opponent Transfer Starts")
+
+        self.usedReadingCards.cards.append(self.curReadingCard)
+        self.removeCurrentGrabbingCard()
+
         if (self.debugMode):
             print(statement)
+
         self.phase = Phase.OPPONENT_TRANSFER
+        self.displayReadingCard = False
+
+    def findPlayersGrabCardSide(self):
+        for i in range(len(self.grabbingCardsInPlay)):
+            if (self.GPinfo.oppoGrabCardLastWord == self.grabbingCardsInPlay[i].getLastWord()):
+                if (i <= self.numYourGrabCardInPlay):  # in self.yourGrabbingCards
+                    self.GPinfo.oppoGrabCardSide = GrabbingCardSide.YOUR_SIDE
+                elif (i > self.numYourGrabCardInPlay):
+                    self.GPinfo.oppoGrabCardSide = GrabbingCardSide.OPPONENT_SIDE
+            if (self.GPinfo.yourGrabCardLastWord == self.grabbingCardsInPlay[i].getLastWord()):
+                if (i <= self.numYourGrabCardInPlay):  # in self.yourGrabbingCards
+                    self.GPinfo.yourGrabCardSide = GrabbingCardSide.YOUR_SIDE
+                elif (i > self.numYourGrabCardInPlay):
+                    self.GPinfo.yourGrabCardSide = GrabbingCardSide.OPPONENT_SIDE
+
+    # Not validated
+    def opponentAdjust(self):
+        for i in range(self.numCardsToTransfer):
+            random.seed(time.time())
+            x = random.randint(1, self.numOppoGrabCardInPlay)
+            print(self.grabbingCardsInPlay[self.numYourGrabCardInPlay + x - 1].getLastWord())
+            for i in range(len(self.grabbingCardsInPlay)):
```

Comment:
is it possible to make it like
`for card in self.grabbingCardsInPlay`?

Even not, at least this should be `j`, since you have used `i` above